EIPS/eip-4931.md

```diff
@@ -0,0 +1,353 @@
+/// the nearest whole amount of destination tokens returning the excess to the msg.sender address. 
+///  Emits the Downgrade event
+/// @param _to The address the source tokens will be sent to upon completion of the downgrade
+/// @param destinationAmount The amount of destination tokens that will be downgraded 
+function downgrade(address _to, uint256 destinationAmount) external
+```
+
+### Events
+
+#### Upgrade
+
+MUST trigger when tokens are upgraded.
+
+``` solidity
+/// @param _from Address that called upgrade
+/// @param _to Address that destination tokens were sent to upon completion of the upgrade
+/// @param sourceAmount Amount of source tokens that were upgraded
+/// @param destinationAmount Amount of destination tokens sent to the _to address
+event Upgrade(address indexed _from, address indexed _to, uint256 sourceAmount, uint256 destinationAmount)
+```
+
+#### Downgrade (Optional Ext.)
+
+MUST trigger when tokens are downgraded.
+
+``` solidity
+/// @param _from Address that called downgrade
+/// @param _to Address that source tokens were sent to upon completion of the downgrade
+/// @param sourceAmount Amount of source tokens sent to the _to address
+/// @param destinationAmount Amount of destination tokens that were downgraded
+event Downgrade(address indexed _from, address indexed _to, uint256 sourceAmount, uint256 destinationAmount)
+}
+```
+
+## Rationale
+There have been several notable ERC20 upgrades (Ex. Golem: GNT -> GLM) where the upgrade functionality is written directly into the token contracts. We view this as a suboptimal approach to upgrades since it tightly couples the upgrade with the existing tokens. This EIP promotes the use of a third contract to facilitate the token upgrade to decouple the functionality of the upgrade from the functionality of the token contracts. Standardizing the upgrade functionality will allow asset holders and exchanges to write simplified reusable scripts to conduct upgrades which will reduce the overhead of conducting upgrades in the future. The interface aims to be intentionally broad leaving much of the specifics of the upgrade to the implementer, so that the token contract implementations do not interfere with the upgrade process. Finally, we hope to create a greater sense of security and validity for token upgrades by enforcing strict means of disposing of the source tokens during the upgrade. This is achieved by the specification of the  ```upgrade``` method. The agreed upon norm is that burnable tokens shall be burned. Otherwise, tokens shall be effectively burned by being sent to the ```0x00``` address. When downgrade Optional Ext. is implemented, the default is instead to lock source tokens in the upgrade contract to avoid a series of consecutive calls to ```upgrade``` and ```downgrade``` from artificially inflating the supply of either token (source or destination). 
```

Comment:
```suggestion
There have been several notable ERC20 upgrades (Ex. Golem: GNT -> GLM) where the upgrade functionality is written directly into the token contracts. We view this as a suboptimal approach to upgrades since it tightly couples the upgrade with the existing tokens. This EIP promotes the use of a third contract to facilitate the token upgrade to decouple the functionality of the upgrade from the functionality of the token contracts. Standardizing the upgrade functionality will allow asset holders and exchanges to write simplified reusable scripts to conduct upgrades which will reduce the overhead of conducting upgrades in the future. The interface aims to be intentionally broad leaving much of the specifics of the upgrade to the implementer, so that the token contract implementations do not interfere with the upgrade process. Finally, we hope to create a greater sense of security and validity for token upgrades by enforcing strict means of disposing of the source tokens during the upgrade. This is achieved by the specification of the  `upgrade` method. The agreed upon norm is that burnable tokens shall be burned. Otherwise, tokens shall be effectively burned by being sent to the `0x00` address. When downgrade Optional Ext. is implemented, the default is instead to lock source tokens in the upgrade contract to avoid a series of consecutive calls to `upgrade` and `downgrade` from artificially inflating the supply of either token (source or destination). 
```

Single backtick for inline code